tests/unit/plugins/listops/dictionary_operation.cpp

```diff
@@ -0,0 +1,186 @@
+void test_dictionary_operation(std::string const& code,
+    std::string const& expected_str)
+{
+    HPX_TEST_EQ(compile_and_run(code), compile_and_run(expected_str));
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+void test_dict_operation()
+{
+    char const* const code = "list(list(1, 2), list(3, 4))";
+
+    phylanx::ir::range key_value =
+        phylanx::execution_tree::extract_list_value(compile_and_run(code));
+
+    phylanx::execution_tree::primitive dict =
+        phylanx::execution_tree::primitives::create_dict_operation(
+            hpx::find_here(),
+            std::vector<phylanx::execution_tree::primitive_argument_type>{
+                std::move(phylanx::execution_tree::primitive_argument_type{
+                    phylanx::ir::range(key_value)})});
+
+    phylanx::execution_tree::primitive_argument_type temp = dict.eval().get();
+
+    phylanx::ir::dictionary f =
+        phylanx::execution_tree::extract_dictionary_value(temp);
+
+    char const* const code_1 = "list(list(1, 2), list(3, 4))";
+
+    phylanx::ir::range key_value_1 =
+        phylanx::execution_tree::extract_list_value(compile_and_run(code_1));
+
+    phylanx::execution_tree::primitive dict_1 =
+        phylanx::execution_tree::primitives::create_dict_operation(
+            hpx::find_here(),
+            std::vector<phylanx::execution_tree::primitive_argument_type>{
+                std::move(phylanx::execution_tree::primitive_argument_type{
+                    phylanx::ir::range(key_value_1)})});
+
+    phylanx::execution_tree::primitive_argument_type temp_1 = dict.eval().get();
+
+    phylanx::ir::dictionary f_1 =
+        phylanx::execution_tree::extract_dictionary_value(temp_1);
+
+    HPX_TEST_EQ(f, f_1);
+}
+
+void test_dict_empty_list_operation()
+{
+    char const* const code = "list()";
+
+    phylanx::ir::range key_value =
+        phylanx::execution_tree::extract_list_value(compile_and_run(code));
+
+    bool exception_thrown = false;
+    try
+    {
+        // Must throw an exception
+        phylanx::execution_tree::primitive dict =
+            phylanx::execution_tree::primitives::create_dict_operation(
+                hpx::find_here(),
+                std::vector<phylanx::execution_tree::primitive_argument_type>{
+                    std::move(phylanx::execution_tree::primitive_argument_type{
+                        phylanx::ir::range(key_value)})});
+
+        phylanx::execution_tree::primitive_argument_type temp =
+            dict.eval().get();
+
+        phylanx::ir::dictionary f =
+            phylanx::execution_tree::extract_dictionary_value(temp);
+        HPX_TEST(false);
+    }
+    catch (std::exception const&)
+    {
+        exception_thrown = true;
+    }
+
+    HPX_TEST(exception_thrown);
+}
+
+void test_dict_empty_arg_operation()
+{
+    char const* const code = "";
+
+    phylanx::ir::range key_value =
+        phylanx::execution_tree::extract_list_value(compile_and_run(code));
```

Comment:
What is the purpose of compiling an empty string?

Reply:
To test the empty argument. I will revise this test in the new update.